### ISSUE TITLE
Create API documentation section for kernelci.org/docs

### ISSUE THREAD

**Issue by gctucker** [kernelci/kernelci-api#4]
All the KernelCI documentation is being collected into a single place on https://kernelci.org/docs.  The main repository for this static website can be found in https://github.com/kernelci/kernelci-project.  Other projects such as `kernelci-core` have a documentation directory that is then imported by `kernelci-project` using a Git submodule and a symlink:

Submodule for `kernelci-core`: https://github.com/kernelci/kernelci-project/tree/main/kernelci.org/external
Relative symlink for kernelci-core: https://github.com/kernelci/kernelci-project/blob/main/kernelci.org/content/en/docs/core

This way, documentation files from other repositories can be pulled into the `content` directory of the main static website repository and the pages are all generated in one place.  We should do the same with `kernelci-api` in a new API section to be able to then start writing actual documentation about it.

Steps:
- [ ] Set up a local instance of the kernelci.org project following the instructions on the [README](https://github.com/kernelci/kernelci-project#kernelciorg-static-website)
- [ ] Add a Git submodule for kernelci-api in the external directory
- [ ] Add a symlink from the content directory to a `doc` directory in the kernelci-api source tree
- [ ] Create a `doc` directory in `kernelci-api` with just an `_index.md` file to provide a starting point for the API section

The new section should appear on the local website instance with the contents from the index page.

To complete this task:
1. Create a commit on a branch for `kernelci-api` adding the `doc/_index.md` file
2. Push this branch to a fork of kernelci-api and create a pull request.
3. Once this has been merged, create a commit on a branch for `kernelci-project` with the git submodule and symlink for kernelci-api
4. Push this branch to a fork of kernelci-project and create a pull request

The new API section should initially appear on https://static.staging.kernelci.org/docs/.  Then once the PR has been merged, it will appear after the next production update on https://kernelci.org/docs

**Comment by JenySadadia** [comment]
Please review the PR. @gctucker

**Comment by JenySadadia** [comment]
Yes, I have committed that change on dev branch on forked kernelci-api. @gctucker
